### ISSUE TITLE
[prefer-spread] falsely fixes string.concat()

### ISSUE THREAD

**Issue by the-ult** [sindresorhus/eslint-plugin-unicorn#2041]
`prefer-spread` falsely marks `string.concat()` to be fixed for spread:

```ts
let test = 'foo';
test = test.concat('bar');
//          ^ Prefer the spread operator over `Array#concat(…)`.eslint[unicorn/prefer-spread]
```

results in:

```ts
 let test = 'foo';
 test = [...test, 'bar'];
// ^  Type 'string[]' is not assignable to type 'string'.
```

@related
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1064#issuecomment-990158420_

**Comment by binury** [comment]
It seems like the `prefer-spread` will always unavoidably generate false positives as the rule is _only_ able to see that a method `concat` is being called on _something_. 
The _only_ checks the rule can make are when the callee is a literal expression like `"/".concat("home")` vs `[].concat("home")`
But the AST contains nothing to make that distinction when the callee is an identifier.
https://astexplorer.net/#/gist/16cf9fb655b23d906794a1fe8e899ef4/latest

**Comment by binury** [comment]
Are we overlooking a solution @fisker ?

**Comment by fregante** [comment]
This looks like a bug, but I'd also never use `String#concat`:

```js
let test = 'foo';
test = test.concat('bar'); // ❌ 
test = test + 'bar'; // ✅ 
test = `${test}bar`; // ✅ 
test = [test, 'bar'].join(''); // ✅ if more than one
```
